feature-libs/product-configurator/rulebased/components/group-menu/configurator-group-menu.component.ts

```diff
@@ -101,18 +101,23 @@ export class ConfiguratorGroupMenuComponent {
 
   click(group: Configurator.Group): void {
     this.configuration$.pipe(take(1)).subscribe((configuration) => {
-      if (!this.configuratorGroupsService.hasSubGroups(group)) {
-        this.configuratorGroupsService.navigateToGroup(configuration, group.id);
-        this.hamburgerMenuService.toggle(true);
-
-        this.configUtils.scrollToConfigurationElement(
-          '.VariantConfigurationTemplate'
-        );
-      } else {
-        this.configuratorGroupsService.setMenuParentGroup(
-          configuration.owner,
-          group.id
-        );
+      if (configuration.interactionState?.currentGroup !== group.id) {
```

Comment:
Wouldn't it be nicer to have a positive if and leave the method?
if (configuration.interactionState?.currentGroup === group.id) {
return;
}
Then we would not have that deep of a nesting

Reply:
Done.